tests/unittest_manager.py

```diff
@@ -325,6 +325,24 @@ def hook(modname: str):
             with self.assertRaises(AstroidBuildingError):
                 self.manager.ast_from_module_name("foo.bar.baz")
 
+    def test_same_name_import_module(self) -> None:
+        """test inference of an import statement with the same name as the module.
+
+        see https://github.com/pycqa/pylint/issues/7289.
```

Comment:
This seems like it's fixing PyCQA/pylint#5151, not PyCQA/pylint#7289, which is about modules that _do not_ end in `.py`. This test case ends in `.py`.

Reply:
I mean, they are both issues of the same type. Basically any file that conflicts with `same-named file and external module` will create this issue.

Want me to add them both? Or change to `5151`?

Reply:
I'm not sure they're the same issue. I thought timmartin demonstrated there was nothing wrong on #5151.

Reply:
In any case, we need a test with a file not ending in .py if we're claiming to fix it! :D